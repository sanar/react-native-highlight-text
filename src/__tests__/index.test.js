import React from 'react';
import { Text } from 'react-native';
import { render } from '@testing-library/react-native';

import HighlightText from '../';

describe('<HighlightText />', () => {
  it('should wrap each matched search word with an isolated element', () => {
    const props = {
      searchWords: ['hello', 'world'],
      textToHighlight: 'In this text we can do a hello to the entire world',
    };
    const { queryByText, toJSON } = render(<HighlightText {...props} />);

    for (const word of props.searchWords) {
      expect(queryByText(word)).toBeTruthy();
    }

    expect(toJSON()).toMatchSnapshot();
  });

  it('should detect all ocurrencies of searched words in textToHighlight prop', () => {
    const props = {
      searchWords: ['Slim', 'Shady'],
      textToHighlight:
        "I'm Slim Shady, yes, I'm the real Shady. All you other Slim Shadys, are just imitating. So won't the real Slim Shady, please stand up?",
    };
    const { getAllByText, toJSON } = render(<HighlightText {...props} />);

    expect(getAllByText('Slim')).toHaveLength(3);
    expect(getAllByText('Shady')).toHaveLength(4);

    expect(toJSON()).toMatchSnapshot();
  });

  it('should render properly with custom styles', () => {
    const props = {
      searchWords: ['hello', 'world'],
      textToHighlight: 'In this text we can do a hello to the entire world',
      style: { color: 'darkgray' },
      highlightStyle: { color: 'blue' },
    };

    const { queryByText, toJSON } = render(<HighlightText {...props} />);

    for (const word of props.searchWords) {
      expect(queryByText(word)).toHaveStyle(props.highlightStyle);
    }

    expect(toJSON()).toMatchSnapshot();
  });

  it('should render properly with custom components', () => {
    const CustomText = (props) => (
      <Text style={{ color: '#333' }}>{props.children}</Text>
    );
    const CustomHighlight = (props) => (
      <Text style={{ fontWeight: 'bold' }}>{props.children}</Text>
    );
    const { toJSON } = render(
      <HighlightText
        searchWords={['hello', 'world']}
        textToHighlight={'In this text we can do a hello to the entire world'}
        textComponent={CustomText}
        highlightComponent={CustomHighlight}
      />
    );
    expect(toJSON()).toMatchSnapshot();
  });
});
