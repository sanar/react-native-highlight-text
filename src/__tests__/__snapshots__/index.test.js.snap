// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HighlightText /> should detect all ocurrencies of searched words in textToHighlight prop 1`] = `
<Text>
  I'm 
  <Text>
    Slim
  </Text>
   
  <Text>
    Shady
  </Text>
  , yes, I'm the real 
  <Text>
    Shady
  </Text>
  . All you other 
  <Text>
    Slim
  </Text>
   
  <Text>
    Shady
  </Text>
  s, are just imitating. So won't the real 
  <Text>
    Slim
  </Text>
   
  <Text>
    Shady
  </Text>
  , please stand up?
</Text>
`;

exports[`<HighlightText /> should render properly with custom components 1`] = `
<Text
  style={
    Object {
      "color": "#333",
    }
  }
>
  In this text we can do a 
  <Text
    style={
      Object {
        "fontWeight": "bold",
      }
    }
  >
    hello
  </Text>
   to the entire 
  <Text
    style={
      Object {
        "fontWeight": "bold",
      }
    }
  >
    world
  </Text>
</Text>
`;

exports[`<HighlightText /> should render properly with custom styles 1`] = `
<Text
  style={
    Object {
      "color": "darkgray",
    }
  }
>
  In this text we can do a 
  <Text
    style={
      Object {
        "color": "blue",
      }
    }
  >
    hello
  </Text>
   to the entire 
  <Text
    style={
      Object {
        "color": "blue",
      }
    }
  >
    world
  </Text>
</Text>
`;

exports[`<HighlightText /> should wrap each matched search word with an isolated element 1`] = `
<Text>
  In this text we can do a 
  <Text>
    hello
  </Text>
   to the entire 
  <Text>
    world
  </Text>
</Text>
`;
